internal/gcsx/random_reader.go

```diff
@@ -353,6 +352,13 @@ func (rr *randomReader) startRead(
 		end = int64(rr.object.Size)
 	}
 
+	// To avoid overloading GCS and to have reasonable latencies, we will only
```

Comment:
Maybe also update the comment on line 324 and 308? Also I think the variable 'size' on 313 (input argument) is unused in this function? Can we call that out too?

Reply:
done.